### PR TITLE
Detect php deprecated language syntax

### PR DESCRIPTION
**Pull request by apm963**
Generate lint warning when `php -l` reports deprecated language syntax.

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -100,12 +100,13 @@ export default {
         const messages = [];
         let match = parseRegex.exec(output);
         while (match !== null) {
-          const line = Number.parseInt(match[2], 10) - 1;
+          const line = Number.parseInt(match[3], 10) - 1;
+          const errorType = match[1];
           messages.push({
-            type: 'Error',
+            type: (errorType.match(/error/i) !== null ? 'Error' : 'Warning'),
```
Comment (Arcanemagus) [inline lib/main.js]: Use `/error/i.test(errorType)` so you get a Boolean directly.

File: spec/linter-php-spec.js [hunk 2]
```diff
@@ -69,4 +70,15 @@ describe('The php -l provider for Linter', () => {
     expect(messages[0].filePath).toBe(fatalPath);
     expect(messages[0].range).toEqual([[10, 4], [10, 25]]);
   });
+
+  it('handles deprecated errors', async () => {
+    const editor = await atom.workspace.open(deprecatedPath);
+    const messages = await lint(editor);
+
+    expect(messages[0].type).toBe('Warning');
+    // PHP 7.x
```
Comment (Arcanemagus) [inline spec/linter-php-spec.js]: You'll need to check what version of PHP is running in the CI environment (maybe there is an environment variable?) and test appropriately as the CI environment checks both 5.x and 7.x versions. You might need to enhance the Travis CI script with a variable of your own to do this if there isn't a default one.

(Apparently missed adding this to the previous review)
It should default to 7.x for when the tests are ran manually outside of the CI environment.

Comment (apm963) [reply spec/linter-php-spec.js]: Thank you for reviewing. I have been looking into ways to check the version of php without having to update the CI environment with something hard-coded. I will update once I get this working.

**Comment by Arcanemagus** [comment]
Did the single file you had before not work for PHP 5? Can you add a separate file testing it under that version?

**Comment by apm963** [comment]
Unfortunately I couldn't make a file that would cause the PHP 5.6 linter to throw a deprecated warning. I tried using the functions and syntax listed in the [deprecated function migration guide](http://php.net/manual/en/migration53.deprecated.php) for versions 5.3 - 5.6 but couldn't find anything that would raise an error with the linter (they would, however, be reported at runtime).

**Comment by Arcanemagus** [comment]
It's possible that PHP 5 just doesn't report deprecations, it is PHP 5 after all 😉.

**Pull request merged**